lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestCloseTask.cs

```diff
@@ -0,0 +1,171 @@
+                .Set(DriverConfiguration.OnTaskRunning, GenericType<StopTaskHandlers>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<StopTaskHandlers>.Class)
+                .Build();
+
+            return TangFactory.GetTang().NewConfigurationBuilder(helloDriverConfiguration).Build();
+        }
+
+        private sealed class StopTaskHandlers :
+            IObserver<IDriverStarted>,
+            IObserver<IAllocatedEvaluator>,
+            IObserver<IActiveContext>,
+            IObserver<ICompletedTask>,
+            IObserver<IRunningTask>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+            private int _contextNumber = 0;
+            private int _taskNumber = 0;
+
+            [Inject]
+            private StopTaskHandlers(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(1).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID" + _contextNumber++)
+                        .Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                value.SubmitTask(GetTaskConfiguration());
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                // Log on task completion to signal a passed test.
+                Logger.Log(Level.Info, CompletedValidationMessage + "Task completed: " + value.Id);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "Task running: " + value.Id);
+                value.Dispose(Encoding.UTF8.GetBytes(DisposeMessageFromDriver));
+            }
+
+            private IConfiguration GetTaskConfiguration()
+            {
+                return TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, "TaskID" + _taskNumber++)
+                    .Set(TaskConfiguration.Task, GenericType<TestCloseTask.StopTestTask>.Class)
+                    .Build();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private sealed class StopTestTask : ITask
+        {
+            [Inject]
+            private StopTestTask()
+            {
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                Thread.Sleep(5 * 1000);
```

Comment:
How would this sleep be disrupted by the call to `Dispose` on the Driver? Also, this test could pass by timeout alone. Further, it doesn't actually validate that the parameter passed to `IRunningTask.Dispose` above makes it to the other side.